core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -149,7 +150,7 @@ private[spark] class HighlyCompressedMapStatus private (
     private[this] var numNonEmptyBlocks: Int,
     private[this] var emptyBlocks: RoaringBitmap,
     private[this] var avgSize: Long,
-    private var hugeBlockSizes: Map[Int, Byte])
+    private[this] var hugeBlockSizes: mutable.Map[Int, Byte])
```

Comment:
this shouldn't be changed, we should still have an immutable map here

Reply:
What about to use more generic scala.collection.Map[Int, Byte] type here?

Reply:
It's mutated though, and needs to be mutable. If it were exposed outside the class, or there was significant danger of accidentally mutating it elsewhere, I think it might be necessary to wrap the result in an immutable wrapper, but here this seems OK to me.

Reply:
I don't think we should change it. Despite we are using a mutable map in order to build it, the result should be an immutable map, as it enforces correctness, avoiding potential bad updates. So I don't think this should be changed. You can just call `toMap` on the mutable one, but in this case, I think the performance would become again like the original one, as there is no override of `toMap` in scala. A better option would then probablt be using a `immutable.Map.Builder`. Could you check it please?

Reply:
I like that possibility. If the cost of wrapping/building an immutable map isn't higher, it's better.

Reply:
Basically, it means that we use immutable Map instead of mutable one, with worse performance characteristics, since MapBuilder just updates inner reference to immutable Map after each insert.  To enforce correctness, what about to use Scala.collection.Map[Int, Byte]? It doesn't allow dangerous mutations operations and mutable.Map is its subtype.

Reply:
yes, you're right. I just checked the implementation of the `MapBuilder`... Seems like there is no efficient way to build an immutable map... I am fine with the approach you're proposing using `scala.collection.Map`, but still seems a bit hacky to me.